### PR TITLE
Php oauth2

### PR DESCRIPTION
**Pull request by vfalco02**
Update php scripts to use Oauth2 by using bearer token in authorization header.

Update `get-access-token` script to add ability to refresh tokens as well.

### CODE REVIEW

File: php/get-access-token [hunk 1]
```diff
@@ -1,64 +1,106 @@
...
-    'consumer_secret' => $consumerSecret,
-    'token' => $params['oauth_token'],
-    'token_secret' => $params['oauth_token_secret'],
-]);
-$stack->remove($requestMiddleware);
-$stack->push($accessMiddleware);
-$res = $client->post('access_token', ['form_params' => ['oauth_verifier' => $verifier]]);
-$credentials = [];
-parse_str($res->getBody(), $credentials);
+    $redirectUri = 'https://localhost';
```
Comment (amberheilman) [inline php/get-access-token]:
```suggestion
    $redirectUri = 'http://localhost';
```

File: php/get-access-token [hunk 2]
```diff
@@ -1,64 +1,106 @@
...
-$params = [];
-parse_str($res->getBody(), $params);
+echo 'Do you wish to create(c) tokens or refresh(r) tokens? ';
+$createRefresh = rtrim(fgets(STDIN), PHP_EOL);
 
-// Log in to receive a verifier code for the request token
-echo "Go to this url: " . OAUTH_URL . "authorize?oauth_token={$params['oauth_token']}\n";
-echo 'Log in and paste the returned verifier code here: ';
-$verifier = rtrim(fgets(STDIN), PHP_EOL);
+if (strtoupper($createRefresh) == 'C') {
```
Comment (amberheilman) [inline php/get-access-token]:
```Fatal error: Uncaught League\OAuth2\Client\Provider\Exception\IdentityProviderException: invalid_request in public-api-examples/php/vendor/league/oauth2-client/src/Provider/GenericProvider.php:222
Stack trace:
#0 public-api-examples/php/vendor/league/oauth2-client/src/Provider/AbstractProvider.php(628): League\OAuth2\Client\Provider\GenericProvider->checkResponse(Object(GuzzleHttp\Psr7\Response), Array)
#1 public-api-examples/php/vendor/league/oauth2-client/src/Provider/AbstractProvider.php(537): League\OAuth2\Client\Provider\AbstractProvider->getParsedResponse(Object(GuzzleHttp\Psr7\Request))
#2 public-api-examples/php/get-access-token(57): League\OAuth2\Client\Provider\AbstractProvider->getAccessToken(Object(League\OAuth2\Client\Grant\AuthorizationCode), Array)
#3 {main}
  thrown in public-api-examples/php/vendor/league/oauth2-client/src/Provider/GenericProvider.php on line 222```

File: php/get-access-token (hunk 1 above)
Comment (vfalco02) [reply php/get-access-token]: We require the redirect uri to be https

File: php/get-access-token (hunk 2 above)
Comment (vfalco02) [reply php/get-access-token]: I cannot reproduce this

File: php/get-access-token [hunk 3]
```diff
@@ -1,64 +1,106 @@
...
-]);
-$stack->remove($requestMiddleware);
-$stack->push($accessMiddleware);
-$res = $client->post('access_token', ['form_params' => ['oauth_verifier' => $verifier]]);
-$credentials = [];
-parse_str($res->getBody(), $credentials);
+    $redirectUri = 'https://localhost';
+
+    $scopes = array(
+        'account.read',
```
Comment (amberheilman) [inline php/get-access-token:23]: im wondering if we should make this configurable?

Comment (vfalco02) [reply php/get-access-token:23]: Team decision was to just leave it hardcoded.

**Comment by amberheilman** [comment]
👍

**Pull request merged**